src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -680,7 +680,9 @@ protected virtual ExpressionToken NextTokenImplementation(out Exception error)
                         }
 
                         int start = this.textPos;
-                        this.ParseIdentifier();
+
+                        // Include dots for the case of annotation.
+                        this.ParseIdentifier(true /*includingDots*/);

```

Comment:
>true /*includingDots*/ [](start = 45, length = 22)

So we allow dots for either the parameter case or the annotation case.  Dots aren't really allowed in parameter names, but for the purpose of parsing the expression it is probably fine to include them as part of the identifier (i.e., there's not a valid case where a dot would be a valid separator between a parameter and something else, so including it in the identifier is reasonable)

Reply:
Yes, should be OK to include dots, which is further used to differentiate parameter alias and annotation per OData protocol. Please let me know if I miss anything in your comment.

---
In reply to: [176545588](https://github.com/OData/odata.net/pull/1123#discussion_r176545588) [](ancestors = 176545588)